src/NuGet.Status/Controllers/AdminController.cs

```diff
@@ -3,94 +3,115 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
 using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 using System.Web.Mvc;
 using Microsoft.WindowsAzure.Storage.Table;
 using NuGet.Services.Status.Table.Manual;
+using NuGet.Status.Configuration;
 using NuGet.Status.Helpers;
 using NuGet.Status.Models;
 using NuGet.Status.Utilities;
 
 namespace NuGet.Status.Controllers
 {
-    [Authorize(Roles = "Admins")]
+    [Authorize]
     public class AdminController : AppController
     {
         public const string UpdatedTempDataKey = "Updated";
 
+        private string[] adminIdentities;
+        private string[] AdminIdentities => (adminIdentities = adminIdentities ?? 
```

Comment:
Maybe just do this in a constructor. I think that's a bit more readable than a lazy-like thing.

Reply:
what would be the error behavior if the AdminEntities config is malformed? Please make sure we fail fast (during service init with a clear error message

---
In reply to: [513149981](https://github.com/NuGet/NuGet.Services.Status/pull/22#discussion_r513149981) [](ancestors = 513149981)

Reply:
@skofman1 It's a ';'-delimited string, so the only error would be a false negative where an identity isn't found through wrong use of a delimiter or a typo. Misconfiguring would also include entries not conforming to an email pattern. I could put some intelligence in there to determine whether this is the case if you feel it's appropriate.

Reply:
i think the check you added is sufficient.